src/test/euler/staking/IntegrationStakeableEulerERC4626.t.sol

```diff
@@ -0,0 +1,165 @@
+        uint256 balanceInContract = eTokenBalance - stakedBalance;
+        _stake(stakedBalance);
+        assertEq(vault.totalAssets(), _toUnderlying(aliceShares + bobShares), 'Total Assets 4');
+        assertEq(eToken.balanceOf(address(vault)), balanceInContract, 'Vault eToken Balance 2');
+        assertEq(stakingRewards.balanceOf(address(vault)), stakedBalance, 'Staked eToken Balance 1');
+        
+        // Simulate that a day has passed
+        vm.warp(block.timestamp + 1 days);
+
+        // Check reward
+        (address rewardsToken, uint256 earned) = vault.reward();
+        assertGt(earned, 0);
+
+        // Claim reward
+        _claim(recipient);
+        assertEq(ERC20(rewardsToken).balanceOf(recipient), earned);
+        assertEq(ERC20(rewardsToken).balanceOf(address(vault)), 0);
+        (, uint256 emptyEarned) = vault.reward();
+        assertEq(emptyEarned, 0);
+
+        // Withdraw all of Alice's balance, so that an unstake is necessary
+        uint256 underlyingInContract = eToken.balanceOfUnderlying(address(vault));
+        uint256 newAliceUnderlying = vault.previewRedeem(aliceShares);
+        uint256 neededToUnstake = eToken.convertUnderlyingToBalance(newAliceUnderlying - underlyingInContract);        
+        uint256 extraWithdrew = FixedPointMathLib.mulDivUp(stakedBalance - neededToUnstake, 5, 100);
+        uint256 leftStaking = stakedBalance - neededToUnstake - extraWithdrew;
+
+        vm.prank(alice);
+        vault.redeem(aliceShares, alice, alice);
+        uint256 eTokenBalance2 = eToken.balanceOf(address(vault));
+        
+        assertEq(underlying.balanceOf(alice), newAliceUnderlying, 'Alice Balance');
+        assertEq(stakingRewards.balanceOf(address(vault)), leftStaking, 'Staked eToken Balance 2');
+        // This is a special check. Due to precision, we might be left with more eTokens thatn expected, or 
+        // we might end up with one wei less than the 5% extra withdrawn
+        assertGe(eTokenBalance2, extraWithdrew - 1, 'Vault eToken Balance 3'); 
+
+        // Unstake a portion
+        uint256 unstake = leftStaking / 2;
+        _unstake(unstake);
+        assertEq(eToken.balanceOf(address(vault)), eTokenBalance2 + unstake);
+        assertEq(stakingRewards.balanceOf(address(vault)), leftStaking - unstake);
+
+        // Simulate that another day has passed
+        vm.warp(block.timestamp + 1 days);
+
+        // Stop staking
+        (, uint256 earned2) = vault.reward();
+        assertGt(earned2, 0);
+        _stopStaking(recipient);
+        assertEq(ERC20(rewardsToken).balanceOf(recipient), earned + earned2);
+        assertEq(ERC20(rewardsToken).balanceOf(address(vault)), 0);
+        (, uint256 emptyEarned2) = vault.reward();
+        assertEq(emptyEarned2, 0);
+        assertEq(address(vault.stakingRewards()), address(0));
+        assertEq(eToken.allowance(address(vault), address(stakingRewards)), 0);
```

Comment:
Maybe we should check `eToken.balanceOf(vault)` = leftStaking + eTokenBalance2 (I think taht would be the correct assert)